test/e2e/framework/networking_utils.go

```diff
@@ -911,13 +974,12 @@ func TestHitNodesFromOutsideWithCount(externalIP string, httpPort int32, timeout
 	hittedHosts := sets.NewString()
 	count := 0
 	condition := func() (bool, error) {
-		var respBody bytes.Buffer
-		reached, err := TestReachableHTTPWithContentTimeout(externalIP, int(httpPort), "/hostname", "", &respBody,
-			1*time.Second)
-		if err != nil || !reached {
+		result := PokeHTTP(externalIP, int(httpPort), "/hostname", &HTTPPokeParams{Timeout: 1 * time.Second})
+		if result.Code != 0 {
 			return false, nil
```

Comment:
This is wrong; it will return and retry on `result.Code == 200`, meaning there's no way to ever update the counts below and return `true`, meaning the function will always time out. (You probably meant `==` rather than `!=` here, although `if result.Status != HTTPOK` would be closer to the original code; you don't want to check the body if `Status == HTTPWrongCode`). 

It looks like this function only gets used from `It("[Slow] [Serial] should create valid firewall rules for LoadBalancer type service"` in `test/e2e/network/firewall.go`, which I guess isn't getting run by CI because it's Slow?

I guess you probably want to force a run with *all* Slow networking tests, to fully test the new code

Reply:
Nice catch.